backtrack/general_solution.md

```diff
@@ -1,156 +1,294 @@
+
+        # Mark
+        visited[idx] = True
+        temp.append(nums[i]);
+
+        # Recurse
+        backtrack(res, temp, nums, visited);
+
+        # Unmark
+        temp.pop()
+        visited[idx] = False
+```
+
+## Permutations II
+(contains duplicates)
+https://leetcode.com/problems/permutations-ii/
+```python
+
+def permute_with_duplicates(nums):
+        """
+        :type nums: List[int]
+        :rtype: List[List[int]]
+        """
+        res = []
+        temp = []
+        visited = [False for _ in range(len(nums))]
+        nums.sort()
+        backtrack(nums, len(nums), visited, temp, res)
+        return res
+
+def backtrack(nums, lnums, visited, temp, res):
+    if len(temp) == lnums:
+        ccopy = [ c for c in temp]
+        res.append(ccopy)
+        return
+    for idx in range(lnums):
+        if visited[idx]:
+            continue
+        if idx and nums[idx-1] == nums[idx] and not visited[idx-1]:
+            continue
+
+        # Mark
+        visited[idx] = True
+        temp.append(nums[idx])
+
+        # Recurse
+        backtrack(nums, lnums, visited, temp, res)
+
+        # Unmark
+        temp.pop()
+        visited[idx] = False
+```
+
+## Combination Sum I
+https://leetcode.com/problems/combination-sum/
+```python
+def combination_sum(nums, target):
+        """
+        :type nums: List[int]
+        :type target: int
+        :rtype: List[List[int]]
+        """
+        res = []
+        temp = []
+        helper(nums, target, 0, temp, res)
+        return res
+        
+def helper(nums, target, pos, temp, res):
+    csum = sum(temp)
+    if target < csum:
+        return
+    if target == csum:
+        res.append(temp[:]) # creating a copy of temp and appending
+        return
+    for idx in range(pos, len(nums)):
+        # Mark
+        temp.append(nums[idx])
+
+        # Recurse
+        helper(nums, target, idx, temp, res)
+
+        # Unmark
+        temp.pop()
+```
+
+## Combination Sum II
+(can't reuse same element)
+https://leetcode.com/problems/combination-sum-ii/
+```python
+def combinationSum2(nums, target):
+        """
+        :type nums: List[int]
+        :type target: int
+        :rtype: List[List[int]]
+        """
+        res = []
+        temp = []
+        nums.sort()
+        backtrack(nums, target, 0, temp, 0, res)
+        return res
+
+def backtrack(nums, target, pos, temp, csum, res):
+    csum = sum(temp)
+    if target < csum:
+        return
+    if target == csum:
+        res.append(temp[:]) # creating a copy of temp and appending
+        return
+    for idx in range(pos, len(nums)):
+        if idx > pos and nums[idx] == nums[idx-1]:
+            continue
+
+        # Mark
+        temp.append(nums[idx])
+
+        # Recurse
+        helper(nums, target, idx+1, temp, csum+nums[idx], res)
+
+        # Unmark
+        temp.pop()
+```
+
+## Palindrome Partitioning
+https://leetcode.com/problems/palindrome-partitioning/
+```python
+def palindrome_partition(s):
```

Comment:
I don't understand the need of these codes here again. Aren't these already implemented in the main algorithm files?